File: src/options.swift
class Options {
  static let sharedInstance = Options()

  var isVerbose = false
  var isShowVersion = false
  var suppliedBranch: String?
  var isBranchSupplied: Bool { return suppliedBranch != nil }
  var isHelp = false
  var isTestRebase = false
  var isShowList = false
  var preferLocal = false
  var preferRemote = false

  func loadOptions(fromArguments arguments: [String]) {
    if arguments.contains("help") || arguments.contains("--help") {
      isHelp = true
    }

    if arguments.contains("--verbose") {
      isVerbose = true
    }

    if arguments.contains("-v") || arguments.contains("--version") {
      isShowVersion = true
    }

    if arguments.contains("--test-rebase") {
      isTestRebase = true
    }

    if arguments.contains("--list") || arguments.contains("-l") {
      isShowList = true
    }

    if arguments.contains("--prefer=local") {
      preferLocal = true
    }

    if arguments.contains("--prefer=remote") {
      preferRemote = true
    }

    if arguments.count > 1 {
      if !arguments[1].hasPrefix("-") {
        suppliedBranch = arguments[1]
      }
    }
  }
}

File: lib/branch_cli/options.rb
class Options < Swifty
  def self.sharedInstance
    @sharedInstance ||= Options.new
  end

  def self.reset
    @sharedInstance = nil
  end

  var isVerbose = false
  var isShowVersion = false
  var suppliedBranch = nil
  var isHelp = false
  var isTestRebase = false
  var isShowList = false
  var preferLocal = false
  var preferRemote = false

  swift self, binding

  def isBranchSupplied
    return suppliedBranch != nil
  end

  def loadOptions(arguments:)
    if arguments.contains("help") || arguments.contains("--help")
      self.isHelp = true
    end

    if arguments.contains("--verbose")
      self.isVerbose = true
    end

    if arguments.contains("-v") || arguments.contains("--version")
      self.isShowVersion = true
    end

    if arguments.contains("--test-rebase")
      self.isTestRebase = true
    end

    if arguments.contains("--list") || arguments.contains("-l")
      self.isShowList = true
    end

    if arguments.contains("--prefer=local")
      self.preferLocal = true
    end

    if arguments.contains("--prefer=remote")
      self.preferRemote = true
    end

    if arguments.count > 0
      if !arguments[0].hasPrefix("-")
        self.suppliedBranch = arguments[0]
      end
    end
  end
end
